pennylane_orquestra/cli_actions.py

```diff
@@ -0,0 +1,236 @@
+            retrieved
+
+    Kwargs:
+        option (str): The option specified for the ``qe get`` CLI call.
+            Examples include ``workflow`` and ``workflowresult``.
+
+    Returns:
+        list: the list of strings representing the response message of the CLI
+        call
+    """
+    process = subprocess.Popen(
+        ["qe", "get", option, str(workflow_id)], stdout=subprocess.PIPE, universal_newlines=True
+    )
+    return process.stdout.readlines()
+
+
+def qe_submit(filepath, keep_file=False):
+    """Function for submitting a workflow via a CLI call.
+
+    Handling submission response messages was based on using Orquestra API
+    v1.0.0.
+
+    Args:
+        filepath (str): the filepath for the workflow file
+
+    Keyword Args:
+        keep_file=False (bool): whether or not to keep or delete the workflow
+            file after submission
+
+    Returns:
+        str: the ID of the workflow submitted
+
+    Raises:
+        ValueError: if the submission was not successful
+    """
+    process = subprocess.Popen(
+        ["qe", "submit", "workflow", str(filepath)], stdout=subprocess.PIPE, universal_newlines=True
+    )
+    res = process.stdout.readlines()
+
+    details = "".join(res)
+    if "Success" not in details:
+        raise ValueError(res)
+
+    if not keep_file:
+        os.remove(filepath)
+
+    unexpected_resp_msg = "Received an unexpected response after submitting workflow."
+    if isinstance(res, list):
+        try:
+            # Get the workflow ID after submitting a workflow
+            workflow_id = res[1].split()[-1]
+        except IndexError as e:
+            raise ValueError(unexpected_resp_msg) from e
+    else:
+        raise ValueError(unexpected_resp_msg)
+
+    return workflow_id
+
+
+def workflow_details(workflow_id):
+    """Function for getting workflow details via a CLI call.
+
+    Args:
+        workflow_id (str): the workflow id for which information will be
+            retrieved
+
+    Returns:
+        list: the list of strings representing the response message of the CLI
+        call
+    """
+    return qe_get(workflow_id, option="workflow")
+
+
+def workflow_results(workflow_id):
+    """Function for getting workflow results via a CLI call.
+
+    Args:
+        workflow_id (str): the workflow id for which information will be
+            retrieved
+
+    Kwargs:
+        option (str): The option specified for the ``qe get`` CLI call.
+            Examples include ``workflow`` and ``workflowresult``.
+
+    Returns:
+        list: the list of strings representing the response message of the CLI
+        call
+    """
+    return qe_get(workflow_id, option="workflowresult")
+
+
+def write_workflow_file(filename, workflow):
+    """Write a workflow file given the name of the file.
+
+    This function will create a YAML file with the workflow content. The file
+    is placed into a user-specific data folder specified by using
+    ``appdirs.user_data_dir``.
```

Comment:
Out of curiosity, what is the reason for writing to the user data directory, as opposed to allowing the directory to be specified solely through the `filename` argument?

Reply:
This function is going to be called by the device, thought it's good to place any files preserved into a well-defined directory that makes sense (instead of e.g., `cwd`).